titus-api/src/main/java/com/netflix/titus/api/jobactivity/service/JobActivityPublisherMetrics.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.api.jobactivity.service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import com.netflix.spectator.api.BasicTag;
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.Tag;
+import com.netflix.titus.api.jobactivity.store.JobActivityPublisherRecord;
+
+public class JobActivityPublisherMetrics {
+
+    private static final String ROOT_NAME = "titus.jobactivity.publish.";
+    private static final String PUBLISH_SUCCESS = ROOT_NAME + "success";
+    private static final String PUBLISH_ERRORS = ROOT_NAME + "errors";
+    private static final String PUBLISH_LATENCY = ROOT_NAME + "latency";
+
+    private static final String RECORD_TYPE_TAG = "type";
+    private static final String RECORD_ID_TAG = "id";
+    private static final String RECORD_REASON_TAG = "reason";
+    private static final String RECORD_OP_TAG = "operation";
+    private static final String RECORD_COUNT_TAG = "count";
+
+    private enum Operations {
```

Comment:
This class looks like a generic RDS metrics collector. Can we generalize it and move to `titus-common`? There is `spectator` package with metric helpers.

Reply:
Done.